source/macos/AdaptiveCards/AdaptiveCards/Managers/ACSVisibilityManager.swift

```diff
@@ -0,0 +1,167 @@
+        let subviews = hostView.stackView.subviews
+        let index = subviews.firstIndex(of: viewToBeHidden) ?? NSNotFound
+        guard index != NSNotFound else { return }
+        let isHead = self.amIHead(index)
+        self.removeVisibleView(at: index)
+        
+        // setting hidden view to hidden again is a programming error
+        // as it requires to have equal or more times of the opposite value to be set
+        // in order to reverse it
+        if !viewToBeHidden.isHidden {
+            viewToBeHidden.isHidden = true
+            // decrease the intrinsic content size by the intrinsic content size of
+            // `viewToBeHidden` otherwise, viewTobeHidden's size will be included
+            hostView.decreaseIntrinsicContentSize(viewToBeHidden)
+            self.changeVisiblityOfAssociatedViews(hostView: viewToBeHidden, visibilityValue: true, contentStackView: hostView)
+        }
+        // if `viewToBeHidden` is a head, get new head if any, and hide its separator
+        if isHead {
+            let headIndex = self.getHeadIndexOfVisibleViews()
+            if headIndex != NSNotFound && headIndex < subviews.count {
+                let headView = subviews[headIndex]
+                changeVisiblityOfSeparator(headView, visibilityHidden: true, contentStackView: hostView)
+            }
+        }
+    }
+    
+    /// unhide `viewToBeUnhidden`. `hostView` is a superview of type ColumnView or ColumnSetView
+    func unhideView(_ viewToBeUnhidden: NSView, hostView: ACRContentStackView) {
+        let subviews = hostView.stackView.subviews
+        let index = subviews.firstIndex(of: viewToBeUnhidden) ?? NSNotFound
+        guard index != NSNotFound else { return }
+        let headIndex = self.getHeadIndexOfVisibleViews()
+        self.addVisibleView(index)
+        let isHead = self.amIHead(index)
+        // check if the unhidden view will become a head
+        if isHead {
+            // only enable filler view associated with the `viewTobeUnhidden`
+            self.changeVisibilityOfPadding(viewToBeUnhidden, visibilityHidden: false)
+            if viewToBeUnhidden.isHidden {
+                viewToBeUnhidden.isHidden = false
+                hostView.increaseIntrinsicContentSize(viewToBeUnhidden)
+            }
+            // previous head view's separator becomes visible
+            if headIndex != NSNotFound && headIndex < subviews.count {
+                let prevHeadView = subviews[headIndex]
+                changeVisiblityOfSeparator(prevHeadView, visibilityHidden: false, contentStackView: hostView)
+            }
+        } else {
+            if viewToBeUnhidden.isHidden {
+                viewToBeUnhidden.isHidden = false
+                hostView.increaseIntrinsicContentSize(viewToBeUnhidden)
+            }
+            self.changeVisiblityOfAssociatedViews(hostView: viewToBeUnhidden, visibilityValue: false, contentStackView: hostView)
+        }
```

Comment:
These 2 have lot of common functionality
Maybe you can use a common util to reduce LOC

Reply:
let me check for this.